### ISSUE TITLE
Parsing error from version 2.0.17 

### ISSUE THREAD

**Issue by emilime** [thymeleaf/thymeleaf#228]
Hello,

we are using Thymeleaf not only for web template but also for email and we have just discovered this strange error:

we use the standard web template engine and in controller a WebContext:

```
final WebContext ctx = new WebContext(request, response, request.getSession().getServletContext(), request.getLocale());
ctx.setVariable("subscriptionDate", new Date());
ctx.setVariable("applicationUrl", RequestUtil.getAppURL(request));

// Create the HTML body using Thymeleaf
final String htmlContent = this.templateEngine.process("mail/site_signup", ctx);
```

in email template this line works until 2.0.16 and gives a parsing error from 2.0.17:

```
<td id="header" class="w640"  bgcolor="#000000" width="640" height="100" valign="bottom" th:style="${'background-image: url(''' + applicationUrl + '/images/ecommerce/mail/beps_header.173309.png''); background-position: center; background-repeat: no-repeat; width:640px; height:100px;'}">
```

the same expression works in a web template but not in email calling templateEngine.process(...)

what can be the problem?

**Comment by jmiguelsamper** [comment]
This error is fixed in 2.0.18.

If you want to avoid the error update to 2.0.18 or 2.0.19 and add this line to your code
        RequestContext requestContext = new RequestContext(request, request.getServletContext());
        ctx.setVariable(SpringContextVariableNames.SPRING_REQUEST_CONTEXT, requestContext);

**Comment by emilime** [comment]
Thank you very much! It works perfectly :)
